vllm/v1/spec_decode/eagle_proposer.py

```diff
@@ -0,0 +1,273 @@
+# SPDX-License-Identifier: Apache-2.0
+import itertools
+
+import torch
+from torch import Tensor, nn
+
+from vllm.config import VllmConfig
+from vllm.forward_context import set_forward_context
+from vllm.v1.attention.backends.flash_attn import FlashAttentionMetadata
+from vllm.v1.outputs import SamplerOutput
+from vllm.v1.sample.metadata import SamplingMetadata
+
+
+class EagleProposer:
+
+    def __init__(self, vllm_config: VllmConfig, model: nn.Module,
+                 sampling_metadata: SamplingMetadata):
+        """
+        Initialize EagleProposer with necessary configuration and model.
+        """
+        self._vllm_config = vllm_config
+        self._model = model
+        self._sampling_metadata = sampling_metadata
+
+    def generate_draft_token_ids(
+            self, *, target_model_input_ids: Tensor,
+            target_model_positions: Tensor, target_model_hidden_states: Tensor,
+            target_model_seq_lens: list[int],
+            sampled_token_ids: list[list[int]],
+            next_prompt_token_ids: list[list[int]], is_prefill: list[bool],
+            num_draft_tokens_to_propose: int,
+            attention_metadata: FlashAttentionMetadata) -> list[SamplerOutput]:
```

Comment:
nit: please append `,` at the end of the input parameters and re-run the formatter, so that each input parameter can take a line. I really recommend this because otw adding/removing an input parameter can change the format again. 
```suggestion
            self,
            *,
            target_model_input_ids: Tensor,
            target_model_positions: Tensor,
            target_model_hidden_states: Tensor,
            target_model_seq_lens: list[int],
            sampled_token_ids: list[list[int]],
            next_prompt_token_ids: list[list[int]],
            is_prefill: list[bool],
            num_draft_tokens_to_propose: int,
            attention_metadata: FlashAttentionMetadata,
        ) -> list[SamplerOutput]:
```